packages/autorest.python/autorest/codegen/models/enum_type.py

```diff
@@ -203,3 +205,7 @@ def imports(self, **kwargs: Any) -> FileImport:
                 else TypingSection.REGULAR,
             )
         return file_import
+
+    @property
+    def enable_generate_model(self) -> bool:
+        return self.code_model.options["models_mode"] and self.enable_generate
```

Comment:
![image](https://user-images.githubusercontent.com/70930885/229698826-655a8878-37fc-4179-af64-a2dfb8c86cfb.png)

`self.code_model.options["models_mode"]` doesn't equal self.models_mode of preprocess: https://github.com/Azure/autorest.python/blob/339978dcafb6e16a467378b02cc64582b303817e/packages/autorest.python/autorest/codegen/__init__.py#L143
https://github.com/Azure/autorest.python/blob/b79615b8b0c28ba2a540f98db7d80dfae1bdc073/packages/autorest.python/autorest/preprocess/__init__.py#L501
https://github.com/Azure/autorest.python/blob/b79615b8b0c28ba2a540f98db7d80dfae1bdc073/packages/autorest.python/autorest/preprocess/__init__.py#L236-L237
And I think current logic is clear and easy to extend in the future